### ISSUE TITLE
Memory leak (multiple GLEngine instances)

### ISSUE THREAD

**Issue by kkl888** [GLWallpaperService/GLWallpaperService#10]
Hi,

Thanks for the amazing opengl wallpaper service. I'm currently developing live wallpaper using JPCT-AE and this GLWallpaperService. It keeps giving out of memory issue after I open and close wallpaper preview for few times (the wallpaper is not set to the phone yet). I analysed it using MAT memory analyser and found out that the created engines are not destroyed even when the onDestroyed() method is called in GLWallpaperService and GLEngine respectively. There are multiple instances of GLEngine kept in context. Any idea what's going on?

**Comment by markfguerra** [comment]
Thanks for reporting this, I haven't seen this one before :-D

Just to troubleshoot, does this bug still occur even using a bare bones wallpaper app? You can try the one built into this repository.

**Comment by kkl888** [comment]
Yes. The image shows the multiple objects using empty Renderer in MAT.
![capture](https://f.cloud.github.com/assets/4495614/637649/afff57ee-d2a8-11e2-99fa-7815708d1761.jpg)

I tried the one from repo, it was still the same.

**Comment by markfguerra** [comment]
I see. I'm not sure how to resolve this at this time, but I'm open to fixes if they come in. If it's alright with you, I would ask you to help me to test those patches if they come in.

**Comment by kkl888** [comment]
Sure. Anything that I can help. Look forward to the fix.
